### ISSUE TITLE
expo prebuild creates xcode workspace and project with underscores removed from the name

### ISSUE THREAD

**Issue by rhoehener** [expo/expo-cli#3552]
is there a specific reason for this?

**Comment by EvanBacon** [comment]
This is probably added to unify the naming across the app, what name are you using?

**Comment by rhoehener** [comment]
`something_app`

My build system modifies some files inside the Xcode project and workspace after `prebuild`, and it failed because of my assumption that those would be named like the `expo.name` value.

**Comment by EvanBacon** [comment]
You can use:
```ts
const { IOSConfig } = require('@expo/config-plugins');

const sourceRoot = IOSConfig.Paths.getSourceRoot(projectRoot);
```

Generally we check for the AppDelegate file and step up from there. Prebuild treats the contents of `ios` and `android` as generated files so preserving the name on generated paths seems like it would create dubious behavior. We strip the underscores to make the name as consistent across the app as possible.

**Comment by rhoehener** [comment]
Not sure I understand. I have `expo.name` in `app.json`. ios folder is cleaned before calling prebuild.

**Comment by rhoehener** [comment]
Would be nice if the slug was used instead of the name for the generated xcodeproj and xcworkspace folders.

**Comment by rhoehener** [comment]
Hopefully the URL-friendliness of the slug is also suitable for XCode project and workspace naming.

**Comment by JaapWeijland** [comment]
I agree. I am using different build configurations, having different names. I want the project and and workspace names to stay the same with all configurations though, so that tools like fastlane know where to find my iOS project regardless of the name.
